scripts/merge_users.py

```diff
@@ -0,0 +1,285 @@
+    LOG.debug("Setting host to %s", mongo_config['host'])
+    LOG.debug("Setting port to %s", mongo_config['port'])
+
+    valid_connection = check_connection(
+        host=mongo_config['host'],
+        port=mongo_config['port'],
+        username=mongo_config['username'],
+        password=mongo_config['password'],
+        authdb=mongo_config['authdb'],
+    )
+
+    LOG.info("Test if mongod is running")
+    if not valid_connection:
+        LOG.warning("Connection could not be established")
+        context.abort()
+
+    try:
+        client = get_connection(**mongo_config)
+    except ConnectionFailure:
+        context.abort()
+
+    database = client[mongo_config['mongodb']]
+
+    LOG.info("Setting up a mongo adapter")
+    mongo_config['client'] = client
+    adapter = MongoAdapter(database)
+
+    user_requests = []
+    event_requests = []
+    acmg_requests = []
+    clinvar_requests = []
+    clinvar_submission_requests = []
+
+    for oi_user_obj in adapter.user_collection.find({'_id': {'$type':'objectId'}}):
+        if not oi_user_obj.get('email'):
+            continue
+
+        LOG.info('===USER===')
+        oi_user_id = oi_user_obj.get('_id')
+        oi_user_email = oi_user_obj.get('email')
+        LOG.info("user: {}".format(oi_user_obj))
+
+        create_alt = False
+        alt_user_obj = adapter.user_collection.find_one({'_id': oi_user_email})
+        if not alt_user_obj:
+            create_alt = True
+            alt_user_obj = copy.deepcopy(oi_user_obj)
+            alt_user_obj['_id'] = oi_user_email
+        else:
+            LOG.info("alt user: {}".format(alt_user_obj))
+            merged_institutes = set()
+            merged_institutes.update(alt_user_obj.get('institutes') + oi_user_obj.get('institutes'))
+            LOG.info('merged institutes: {}'.format(merged_institutes))
+            alt_user_obj['institutes'] = list(merged_institutes)
+
+            merged_roles = set()
+            merged_roles.update(alt_user_obj.get('roles') + oi_user_obj.get('roles'))
+            LOG.info('merged roles: {}'.format(merged_roles))
+            alt_user_obj['roles'] = list(merged_roles)
+
+            created_at = oi_user_obj.get('created_at')
+            alt_created_at = alt_user_obj.get('created_at')
+            if (alt_created_at and alt_created_at < created_at):
+                created_at = alt_created_at
+
+            if created_at:
+                alt_user_obj['created_at'] = created_at
+
+            accessed_at = alt_user_obj.get('accessed_at')
+            oi_accessed_at = oi_user_obj.get('accessed_at')
+            if (oi_accessed_at and oi_accessed_at > accessed_at):
+                accessed_at = oi_accessed_at
+
+            if accessed_at:
+                alt_user_obj['accessed_at'] = accessed_at
+
+        if create_alt:
+            LOG.info("create user: {}".format(alt_user_obj))
+            operation = pymongo.InsertOne(alt_user_obj)
```

Comment:
There is no DryRun clause here!!

Reply:
Meh, this is just appending future operations: bulk write will happen at the end, well protected by dry run clauses. I hope.

Reply:
Haha sorry

Reply:
😅